### ISSUE TITLE
Allow admins to set "trap" names

### ISSUE THREAD

**Issue by Laurelai** [florence-social/mastodon-fork#73]
Allow admins to add words to a configurable list that if a new user tries to make a username that contains that word the account is automatically suspended. Adding a name should also suspend any account that currently exist that contain the word.

This has multiple uses from mass banning similar named bots to preventing people from joining with hate speech in their usernames. 

An option should also exist to whitelist specific words that might contain the trigger word or phrase to prevent wrongful suspensions. Example if you have a legitemate user who has a phrase in their username thats not problematic in itself but a bunch of spambots use that phrase in their name you could whitelist the user and then mass ban the bots.

Another option should also prevent anyone with that phrase from outside of the intstance from interacting with it. The use case for this is one person with many accounts across the fedverse using the same username who is just instance hopping to avoid bans.

**Comment by scunthorpe** [comment]
This is called the [Scunthorpe problem](https://www.youtube.com/watch?v=CcZdwX4noCE), or the penistone problem if you think scunthorpe is too rude.

This is a great way to ban a lot of existing users by mistake.

**Comment by Feufochmar** [comment]
Maybe showing the list of users (or the most active users and the total number of users) that will be automatically banned when adding a blacklisted word could help to avoid mistakes.

**Comment by Laurelai** [comment]
Made an account just for this huh

> This is called the [Scunthorpe problem](https://www.youtube.com/watch?v=CcZdwX4noCE), or the penistone problem if you think scunthorpe is too rude.
> 
> This is a great way to ban a lot of existing users by mistake.

Pretty sure theres no legitemate use case for things like nazi slogans for example in user names.

Also the idea i stated would also have a whitelist for users who match a filer but for non problematic reasons

**Comment by clarfonthey** [comment]
I feel like automatic suspension doesn't make a whole lot of sense for new registrations-- if you would just suspend them, it makes sense to just prevent registration altogether, or make registration invite-only. Automatic filters do make a lot of sense for other instances though.

Presumably this would keyword-filter account handles, names, and bios and flag them automatically, with an option to suspend them automatically.

**Comment by Laurelai** [comment]
Having an automatic suspension can help you filter out bad actors while allowing your instance to keep registration open for good actors, but it might also be good to have an option where instead of automatically suspending them it sends the admin a report for a human to decide what to do.

**Comment by clarfonthey** [comment]
Yeah, basically what I was thinking is that registering and catching a filter would either a) block registering while filter is caught or b) flag an admin to either approve or reject the registration.

When I said "flag" earlier I basically meant add to a list and notify an admin about it so they can take a look when they have time.

**Comment by jhaye** [comment]
I think this could be combined with manual sign-up screening (I don't know if we have an issue for that already).
Basically we could employ the same filter for simply ignoring sign-ups that match.

**Comment by clarfonthey** [comment]
There is actually manual sign-up verification in upstream now, so, it'd be possible.
